### PR TITLE
[Spec] Add invariants on ValidatorSet and StakePool

### PR DESCRIPTION
**Pull request by rahxephon89**
### Description

This PR fixes the daily prover test failure and adds invariants on ValidatorSet and StakePool

### Test Plan

Both CI and daily test pass

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/stake.spec.move [hunk 1]
```diff
@@ -66,6 +69,19 @@ spec aptos_framework::stake {
...
+            old(global<ValidatorSet>(@aptos_framework)) == global<ValidatorSet>(@aptos_framework);
+    }
+
+    spec schema StakePoolNotChangeDuringReconfig {
+        ensures forall a: address where old(exists<StakePool>(a)): reconfiguration_state::spec_is_in_progress() ==>
+            (old(global<StakePool>(a).pending_inactive) == global<StakePool>(a).pending_inactive &&
+            old(global<StakePool>(a).pending_active) == global<StakePool>(a).pending_active &&
+            old(global<StakePool>(a).inactive) == global<StakePool>(a).inactive &&
+            old(global<StakePool>(a).active) == global<StakePool>(a).active);
+    }
```
Comment (zjma) [inline aptos-move/framework/aptos-framework/sources/stake.spec.move:83]: magic!

**Review by danielxiangzl — APPROVED** [review]

**Review by zjma — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/configs/version.spec.move [hunk 2]
```diff
@@ -37,7 +37,7 @@ spec aptos_framework::version {
         pragma verify_duration_estimate = 120;
         include transaction_fee::RequiresCollectedFeesPerValueLeqBlockAptosSupply;
         include staking_config::StakingRewardsConfigRequirement;
-        requires chain_status::is_operating();
+        requires chain_status::is_genesis();
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/configs/version.spec.move:40]: (1) first of all, `set_version` doesn't seem to be a genesis-only function.

(2) `set_version` seems to be deprecated by `set_for_next_epoch()`.

File: aptos-move/framework/aptos-framework/sources/configs/gas_schedule.spec.move [hunk 3]
```diff
@@ -54,11 +54,13 @@ spec aptos_framework::gas_schedule {
         use aptos_framework::aptos_coin::AptosCoin;
         use aptos_framework::transaction_fee;
         use aptos_framework::staking_config;
+        use aptos_framework::chain_status;
 
         // TODO: set because of timeout (property proved)
         pragma verify_duration_estimate = 600;
         requires exists<stake::ValidatorFees>(@aptos_framework);
         requires exists<CoinInfo<AptosCoin>>(@aptos_framework);
+        requires chain_status::is_genesis();
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/configs/gas_schedule.spec.move:63]: `set_gas_schedule` is a genesis-only function?

Moreover, this function seems to be deprecated.

**Review by junkil-park — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**